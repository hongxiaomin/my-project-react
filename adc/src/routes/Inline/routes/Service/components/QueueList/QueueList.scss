.top-left{
    display: flex;
    width: 100%;
    font-size: 16px;
    padding: 0 10px;
    background-color: #f7f7f7;
    border-radius: 4px;
    margin-top: 20px;
    align-items: center;
    justify-content: space-between;
    flex-wrap: wrap;
}

.search-button {
    margin-left: 10px;
}

.reset {
    min-height: 20px;
    line-height: 20px;
    margin-bottom: 3px;
    &:before {
        font-size: 14px;
        margin-bottom: 5px;
    }
}

.filter-elem-w {
  position: relative;
  .filter-elem {
    overflow-x: hidden;
    position: fixed;
    width: 100px;
    text-align: left;
    background: #fff;
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 5px;
    margin-top: 10px;
    z-index: 100;
    max-height: 150px;
    overflow: auto;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.15);
  }
}

.cut-off {
  height: 1px;
  background: #e8e8e8;
  margin: 5px -5px 5px -5px;
}
