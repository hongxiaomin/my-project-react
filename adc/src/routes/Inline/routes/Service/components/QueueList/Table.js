import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import ReactTable from 'components/ADCTable'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import { getQueueStatus } from 'Utils'
import { Button, Popconfirm, Icon, Checkbox } from 'antd'
import { queueAction, getQueueList, resetService } from '../../modules/Service/'
import classes from './QueueList.scss'


const CheckboxGroup = Checkbox.Group
export class Table extends React.Component {
  constructor(props) {
    super(props)

    const { startTime, endTime } = this.formatCondition(this.props.condition)
    this.filterElemOffset = { top: 'auto', left: 'auto' }
    this.state = {
      highlightedIndex: -1,
      expanded: {},
      expandedObject: {},
      startTime,
      endTime,
      pageSize: window.localStorage.getItem('pageSize') || 5, // 每页多少条
      pageNow: 0, // 当前页
      filter: false,
      statusFilterData: getQueueStatus(),
    }
  }

  componentWillReceiveProps(nextProps) {
    const { startTime, endTime } = this.formatCondition(nextProps.condition)
    if (this.state.startTime !== startTime || this.state.endTime !== endTime) {
      // 获取数据
      const { pageSize, pageNow, statusFilterData } = this.state
      this.setState({
        startTime,
        endTime,
      })
      this.props.dispatch(getQueueList({
        pageSize,
        pageNow,
        startTime,
        endTime,
        status: JSON.stringify(statusFilterData),
      }))
    }

    // 计算当前列表展开的位置
    const indexs = []
    const messageIds = Object.keys(this.state.expandedObject)
    const expandedObject = {}

    nextProps.queueList.forEach((item, i) => {
      if (messageIds.includes(item.messageId.toString())) {
        indexs.push(i)
        expandedObject[item.messageId] = 1
      }
    })

    if (indexs.length) {
      const expanded = {}
      indexs.forEach((index) => {
        expanded[index] = {}
      })

      this.setState({
        expanded,
        expandedObject,
      })
    } else {
      this.setState({
        expanded: {},
        expandedObject: {},
      })
    }
  }

  getQueueButtonColor = (job, isLoading) => {
    if (job.messageStatus === 'received' && !isLoading) {
      return '#0099FF'
    }

    return ''
  }

  getGlassButtonColor = (glass, jobStatus, isLoading) => {
    if (glass.glassStatus === 'waiting' && (jobStatus === 'waiting' || jobStatus === 'running') && !isLoading) {
      return '#0099FF'
    }
    return ''
  }

  getGlassColumns = (job) => {
    const { intl } = this.props
    const columnsStyle = { textAlign: 'center' }
    const columns = [
      {
        Header: intl.formatMessage({ id: 'serviceManagement.queueList.glassId' }),
        accessor: 'glassName',
        minWidth: 'auto',
        resizable: false,
        style: { ...columnsStyle },
      },
      {
        Header: intl.formatMessage({ id: 'serviceManagement.queueList.status' }),
        accessor: 'glassStatus',
        minWidth: 'auto',
        width: 80,
        resizable: false,
        style: { ...columnsStyle },
        Cell: row => (
          <div style={{
            width: '100%',
            color: (() => {
              switch (row.value.toLowerCase()) {
                case 'done': return '#66cc00'
                case 'processing': return 'orange'
                default: return ''
              }
            })(),
          }}
          >{row.value}
          </div>
        ),
      },
    ]
    if (this.props.showAction) {
      columns.push({
        Header: intl.formatMessage({ id: 'serviceManagement.queueList.action' }),
        minWidth: 'auto',
        width: 45,
        resizable: false,
        style: { ...columnsStyle },
        Cell: (row) => {
          const isLoading = this.props.loading.includes(`glass_${row.original.glassId}`)
          const color = this.getGlassButtonColor(row.original,
            job.original.messageStatus, isLoading)
          return (<Button
            icon={isLoading ? 'loading' : 'caret-right'}
            title={intl.formatMessage({ id: 'serviceManagement.queueList.startJudging' })}
            disabled={!color}
            style={{
              color,
              border: 'none',
              background: 'none',
            }}
            onClick={this.handeleGlasses(row, isLoading)}
          />)
        },
      })
    }
    return columns
  }

  filterClose = () => {
    this.setState({
      filter: false,
    })
    document.removeEventListener('click', this.filterClose)
  }

  filterToggle = (e) => {
    e.nativeEvent.stopImmediatePropagation()
    this.setState({
      filter: !this.state.filter,
    })
    if (!this.state.filter) {
      document.addEventListener('click', this.filterClose)
    }
  }

  filterChange = (v) => {
    this.setState({
      statusFilterData: v,
    })
  }

  boxChange = (e) => {
    e.nativeEvent.stopImmediatePropagation()
  }

  filterOk = () => {
    const { pageSize, pageNow, startTime, endTime, statusFilterData } = this.state
    this.props.dispatch(getQueueList({
      pageSize,
      pageNow,
      startTime,
      endTime,
      status: JSON.stringify(statusFilterData),
    }))
  }

  formatCondition = condition => ({
    startTime: condition[0],
    endTime: condition[1],
  })

  handleQueue = (row, isLoading) => () => {
    if (row.original.messageStatus !== 'received' || isLoading) return
    this.props.dispatch(queueAction({
      type: 'message',
      id: row.original.messageId.toString(),
    }))
  }

  handeleGlasses = (row, isLoading) => () => {
    if (row.original.glassStatus !== 'waiting' || isLoading) return
    this.props.dispatch(queueAction({
      type: 'glass',
      id: row.original.glassId.toString(),
    }))
  }

  handleExpandedChange = (newExpanded, index) => {
    const expanded = Object.assign({}, newExpanded)
    const expandedObject = Object.assign({}, this.state.expandedObject)
    if (!expanded[index]) {
      delete expandedObject[this.props.queueList[index].messageId]
    } else {
      expandedObject[this.props.queueList[index].messageId] = 1
    }
    this.setState({
      expanded,
      expandedObject,
    }, () => {
      console.log(this.state.expanded, this.state.expandedObject)
    })
  }

  handlePageChange = (pageNow) => {
    // 传入当前的页码数， 还有其他的一些参数
    const { pageSize, startTime, endTime, statusFilterData } = this.state
    this.props.dispatch(getQueueList({
      pageNow,
      pageSize,
      startTime,
      endTime,
      status: JSON.stringify(statusFilterData),
    }))
    this.setState({
      pageNow,
      expanded: {},
      expandedObject: {},
    })
  }

  handlePageSize = (pageSize, pageNow) => {
    window.localStorage.setItem('pageSize', pageSize)
    // 当更改pageSize的时候要注意， 当前页和每页条数， 超过总数的情况
    // 先做成， 改变每页条数， 当前页变成0
    const { startTime, endTime, statusFilterData } = this.state
    this.setState({
      pageNow: 0,
      pageSize,
    })

    this.props.dispatch(getQueueList({
      pageNow: 0,
      pageSize,
      startTime,
      endTime,
      status: JSON.stringify(statusFilterData),
    }))
  }

  handleSorted = (newSorted, column, shiftKey) => {
    // shiftKey 会把排序删掉
    if (shiftKey) return
    const { pageSize, pageNow, startTime, endTime, statusFilterData } = this.state
    const sorted = newSorted[0] || {}
    const field = sorted.id
    const desc = sorted.desc

    this.props.dispatch(getQueueList({
      pageSize,
      pageNow,
      startTime,
      endTime,
      field,
      desc,
      status: JSON.stringify(statusFilterData),
    }))
  }

  cell = name => ({ original }) => <span title={original[name]}>{original[name]}</span>

  reset = row => () => {
    const { pageSize, pageNow, startTime, endTime, statusFilterData } = this.state
    this.props.dispatch(resetService({
      messageId: row.original.messageId.toString(),
    }, {
      pageSize,
      pageNow,
      startTime,
      endTime,
      status: JSON.stringify(statusFilterData),
    }))
  }

  filterReset = () => {
    this.setState({
      statusFilterData: getQueueStatus(),
    }, () => {
      const { pageSize, pageNow, startTime, endTime, statusFilterData } = this.state
      this.props.dispatch(getQueueList({
        pageNow,
        pageSize,
        startTime,
        endTime,
        status: JSON.stringify(statusFilterData),
      }))
    })
  }

  render() {
    const { intl } = this.props
    const fitlerElem = (<div styleName="filter-elem-w">
      <span><WrapperFormattedMessage id="serviceManagement.queueList.status" /></span>
      <Icon
        type="filter"
        onClick={this.filterToggle}
        style={{
          cursor: 'pointer',
          fontSize: 16,
          position: 'relative',
          left: 3,
          top: 4,
        }}
      />
      <div
        styleName="filter-elem"
        style={{
          display: this.state.filter ? 'block' : 'none',
          top: this.filterElemOffset.top + 10,
        }}
        onClick={this.boxChange}
      >
        <CheckboxGroup
          defaultValue={this.state.statusFilterData}
          value={this.state.statusFilterData}
          onChange={this.filterChange}
        >
          {getQueueStatus().map((item, i) => (<p key={item}><Checkbox key={item} value={item}>{item}</Checkbox></p>))}
        </CheckboxGroup>
        <p styleName="cut-off" />
        <a
          onClick={this.filterOk}
          style={{
            float: 'left',
          }}
        >OK</a>
        <a
          onClick={this.filterReset}
          style={{
            float: 'right',
          }}
        ><WrapperFormattedMessage id="reset" /></a>
      </div>
    </div>)
    const columnsStyle = { textAlign: 'center' }
    const queueColumns = [
      {
        Header: intl.formatMessage({ id: 'processId' }),
        accessor: 'processName',
        minWidth: 'auto',
        resizable: false,
        Cell: this.cell('processName'),
        style: { ...columnsStyle },
      },
      {
        Header: intl.formatMessage({ id: 'productId' }),
        accessor: 'productName',
        minWidth: 'auto',
        resizable: false,
        Cell: this.cell('productName'),
        style: { ...columnsStyle },
      },
      {
        Header: intl.formatMessage({ id: 'serviceManagement.dialogFilter.model' }),
        accessor: 'modelName',
        minWidth: 'auto',
        resizable: false,
        Cell: this.cell('modelName'),
        style: { ...columnsStyle },
      },
      {
        Header: fitlerElem,
        accessor: 'messageStatus',
        minWidth: 'auto',
        width: 80,
        resizable: false,
        style: { ...columnsStyle },
        Cell: row => (
          <div style={{
            width: '100%',
            color: (() => {
              switch (row.value.toLowerCase()) {
                case 'done': return '#66cc00'
                case 'received': return 'orange'
                case 'running': return '#0099FF'
                default: return ''
              }
            })(),
          }}
          >{row.value}
          </div>
        ),
      },
      {
        Header: intl.formatMessage({ id: 'serviceManagement.queueList.arrivalTime' }),
        minWidth: 'auto',
        width: 120,
        resizable: false,
        accessor: 'arrivalTime',
        Cell: this.cell('arrivalTime'),
        style: { ...columnsStyle },
      },
    ]

    if (this.props.showAction) {
      queueColumns.push({
        Header: intl.formatMessage({ id: 'userAndLicenseManagement.license.action' }),
        minWidth: 'auto',
        width: 80,
        resizable: false,
        style: { ...columnsStyle },
        Cell: (row) => {
          const isLoading = this.props.loading.includes(`message_${row.original.messageId}`)
          return (<p>
            <Button
              icon={isLoading ? 'loading' : 'caret-right'}
              title={intl.formatMessage({ id: 'serviceManagement.queueList.chooseTask' })}
              disabled={isLoading || !this.getQueueButtonColor(row.original, isLoading)}
              style={{
                color: this.getQueueButtonColor(row.original, isLoading),
                border: 'none',
                background: 'none',
              }}
              onClick={this.handleQueue(row, isLoading)}
            />
            <Popconfirm
              title={intl.formatMessage({ id: 'serviceManagement.queueList.resetData' })}
              okText={intl.formatMessage({ id: 'yes' })}
              cancelText={intl.formatMessage({ id: 'no' })}
              placement="left"
              onConfirm={this.reset(row)}
            >
              <button
                style={{ fontSize: '14px' }}
                type="button"
                className={`pt-button pt-minimal pt-icon-history ${classes.reset}`}
                title={intl.formatMessage({ id: 'reset' })}
              />
            </Popconfirm>
          </p>)
        },
      })
    }

    const { queueList: list } = this.props

    return (
      <div>
        <ReactTable
          manual
          data={list}
          showPageJump
          sortable={false}
          pages={Math.ceil(this.props.queueListLength / this.state.pageSize)}
          className="-striped -highlight"
          defaultPageSize={this.state.pageSize}
          columns={queueColumns}
          expanded={this.state.expanded}
          pageSizeOptions={[5, 10, 15, 20, 30, 40, 50]}
          onExpandedChange={this.handleExpandedChange}
          onPageChange={this.handlePageChange}
          onPageSizeChange={this.handlePageSize}
          onSortedChange={this.handleSorted}
          getTrProps={() => ({ style: { height: '32px' } })}
          SubComponent={row => (<div style={{ padding: '10px' }}>
            <ReactTable
              data={row.original.glassList}
              columns={this.getGlassColumns(row)}
              defaultPageSize={row.original.glassList.length > 5
                ? row.original.glassList.length
                : 5
              }
              showPagination={false}
              getTrProps={() => ({ style: { height: '32px' } })}
              style={{
                height: '195px',
              }}
            />
          </div>)}

          style={{
            maxHeight: '475px',
            height: this.state.pageSize > 10 ? '475px' : 'auto',
          }}
        />
        <p style={{ lineHeight: '40px' }}><WrapperFormattedMessage id='totalNumber' /> {this.props.queueListLength}</p>
      </div>
    )
  }
}

Table.propTypes = {
  queueList: PropTypes.array.isRequired,
  queueListLength: PropTypes.number.isRequired,
  showAction: PropTypes.bool.isRequired,
  condition: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
  loading: PropTypes.array.isRequired,
  intl: PropTypes.object.isRequired,
}

export default connect(
  ({ auth, service, intl: { locale } }) => ({
    locale,
    token: auth.token,
    queueList: service.queueList,
    queueListLength: service.queueListLength,
    loading: service.loading.queue,
  }))(injectIntl(Table))
