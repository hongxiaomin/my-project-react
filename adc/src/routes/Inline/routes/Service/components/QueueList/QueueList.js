/* eslint import/no-named-as-default: 0 */
import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import { Radio, DatePicker } from 'antd'
import Table from './Table'
import { queueModeSwitch, getQueueMode } from '../../modules/Service'
import './QueueList.scss'

const RangePicker = DatePicker.RangePicker
const RadioGroup = Radio.Group
const RadioButton = Radio.Button

export class QueueList extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      // status: 'manualMode', // autoMode
      date: [],
    }
  }

  componentDidMount() {
    this.props.dispatch(getQueueMode())
  }

  onChange = (date, dateString) => {
    this.setState({
      date: dateString,
    })
  }

  handleStatusChange = (e) => {
    this.props.dispatch(queueModeSwitch(e.target.value))
  }

  disabledDate = current => (current && current.valueOf() > Date.now())

  render() {
    const { intl } = this.props
    return (
      <div style={{ width: '50%' }}>
        <h4 style={{ paddingTop: '10px' }}>
          <WrapperFormattedMessage id="serviceManagement.queueList.queueInformation" />
        </h4>
        <div styleName="top-left">
          <div style={{ padding: '10px 0' }}>
            <span style={{ fontSize: '14px' }}>
              <WrapperFormattedMessage id="serviceManagement.queueList.filter" />: </span>
            <RangePicker
              onChange={this.onChange}
              placeholder={[intl.formatMessage({ id: 'startTime' }), intl.formatMessage({ id: 'endTime' })]}
              format="YYYY-MM-DD HH:mm:ss"
              disabledDate={this.disabledDate}
              showTime
            />
          </div>
          <div style={{ padding: '10px 0' }}>
            <RadioGroup value={this.props.queueModel} onChange={this.handleStatusChange}>
              <RadioButton value="manualMode"><WrapperFormattedMessage id="serviceManagement.edit.manualMode" /></RadioButton>
              <RadioButton value="autoMode"><WrapperFormattedMessage id="serviceManagement.edit.autoMode" /></RadioButton>
            </RadioGroup>
          </div>
        </div>
        <Table
          condition={this.state.date}
          showAction={this.props.queueModel === 'manualMode'}
        />
      </div>
    )
  }
}

QueueList.propTypes = {
  queueModel: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
}

export default connect(({ service, intl: { locale } }) => ({
  locale,
  queueModel: service.queueModel,
}))(injectIntl(QueueList))
