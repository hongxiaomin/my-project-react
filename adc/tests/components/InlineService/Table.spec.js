import React from 'react'
import _ from 'lodash'
import { Table } from 'routes/Inline/routes/Service/components/QueueList/Table'

describe('(Component) Table', () => {
  const queueList = []
  const queueListLength = 1
  const showAction = false
  const condition = []
  const loading = []
  const dispatch = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <Table
        queueList={queueList}
        dispatch={dispatch}
        queueListLength={queueListLength}
        showAction={showAction}
        condition={condition}
        loading={loading}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render showAction is not null', () => {
    const wrapper = shallow(
      <Table
        queueList={queueList}
        dispatch={dispatch}
        queueListLength={queueListLength}
        showAction={showAction}
        condition={[{ id: 2 }]}
        loading={loading}
      />,
    )
    wrapper.setState({ filter: 'block' })
    wrapper.setState({ pageSize: 11 })
    // wrapper.find('Icon').simulate('change', { nativeEvent: { stopImmediatePropagation: _.noop } })
    // expect(wrapper.state('filter').toEqual('none'))
    expect(wrapper).toMatchSnapshot()
  })
})
