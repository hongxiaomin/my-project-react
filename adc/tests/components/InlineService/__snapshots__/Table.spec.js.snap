// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Table render normally 1`] = `
<div>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    SubComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className="-striped -highlight"
    collapseOnDataChange={true}
    collapseOnPageChange={true}
    collapseOnSortingChange={true}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Cell": [Function],
          "Header": "Process ID",
          "accessor": "processName",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
        },
        Object {
          "Cell": [Function],
          "Header": "Product ID",
          "accessor": "productName",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
        },
        Object {
          "Cell": [Function],
          "Header": "Model",
          "accessor": "modelName",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
        },
        Object {
          "Cell": [Function],
          "Header": <div
            styleName="filter-elem-w"
      >
            <span>
                  Status
            </span>
            <Icon
                  onClick={[Function]}
                  style={
                        Object {
                              "cursor": "pointer",
                              "fontSize": 16,
                              "left": 3,
                              "position": "relative",
                              "top": 4,
                            }
                  }
                  type="filter"
            />
            <div
                  onClick={[Function]}
                  style={
                        Object {
                              "display": "none",
                              "top": "auto10",
                            }
                  }
                  styleName="filter-elem"
            >
                  <CheckboxGroup
                        defaultValue={
                              Array [
                                    "received",
                                    "receiving",
                                    "running",
                                    "done",
                                    "waiting",
                                  ]
                        }
                        onChange={[Function]}
                        options={Array []}
                        prefixCls="ant-checkbox-group"
                        value={
                              Array [
                                    "received",
                                    "receiving",
                                    "running",
                                    "done",
                                    "waiting",
                                  ]
                        }
                  >
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="received"
                              >
                                    received
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="receiving"
                              >
                                    receiving
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="running"
                              >
                                    running
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="done"
                              >
                                    done
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="waiting"
                              >
                                    waiting
                              </Checkbox>
                        </p>
                  </CheckboxGroup>
                  <p
                        styleName="cut-off"
                  />
                  <a
                        onClick={[Function]}
                        style={
                              Object {
                                    "float": "left",
                                  }
                        }
                  >
                        OK
                  </a>
                  <a
                        onClick={[Function]}
                        style={
                              Object {
                                    "float": "right",
                                  }
                        }
                  >
                        Reset
                  </a>
            </div>
      </div>,
          "accessor": "messageStatus",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
          "width": 80,
        },
        Object {
          "Cell": [Function],
          "Header": "Arrival Time",
          "accessor": "arrivalTime",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
          "width": 120,
        },
      ]
    }
    data={Array []}
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPageSize={5}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanded={Object {}}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    manual={true}
    nestingLevelKey="_nestingLevel"
    nextText="Next"
    noDataText="No rows found"
    ofText="of"
    onExpandedChange={[Function]}
    onFetchData={[Function]}
    onPageChange={[Function]}
    onPageSizeChange={[Function]}
    onSortedChange={[Function]}
    originalKey="_original"
    pageSizeOptions={
      Array [
        5,
        10,
        15,
        20,
        30,
        40,
        50,
      ]
    }
    pageText="Page"
    pages={1}
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="Previous"
    resizable={true}
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={true}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={false}
    style={
      Object {
        "height": "auto",
        "maxHeight": "475px",
      }
    }
    subRowsKey="_subRows"
  />
  <p
    style={
      Object {
        "lineHeight": "40px",
      }
    }
  >
    Total number 
    1
  </p>
</div>
`;

exports[`(Component) Table render showAction is not null 1`] = `
<div>
  <ReactTable
    AggregatedComponent={[Function]}
    ExpanderComponent={[Function]}
    FilterComponent={[Function]}
    LoadingComponent={[Function]}
    NoDataComponent={[Function]}
    PadRowComponent={[Function]}
    PaginationComponent={[Function]}
    PivotValueComponent={[Function]}
    ResizerComponent={[Function]}
    SubComponent={[Function]}
    TableComponent={[Function]}
    TbodyComponent={[Function]}
    TdComponent={[Function]}
    TfootComponent={[Function]}
    ThComponent={[Function]}
    TheadComponent={[Function]}
    TrComponent={[Function]}
    TrGroupComponent={[Function]}
    aggregatedKey="_aggregated"
    className="-striped -highlight"
    collapseOnDataChange={true}
    collapseOnPageChange={true}
    collapseOnSortingChange={true}
    column={
      Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      }
    }
    columns={
      Array [
        Object {
          "Cell": [Function],
          "Header": "Process ID",
          "accessor": "processName",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
        },
        Object {
          "Cell": [Function],
          "Header": "Product ID",
          "accessor": "productName",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
        },
        Object {
          "Cell": [Function],
          "Header": "Model",
          "accessor": "modelName",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
        },
        Object {
          "Cell": [Function],
          "Header": <div
            styleName="filter-elem-w"
      >
            <span>
                  Status
            </span>
            <Icon
                  onClick={[Function]}
                  style={
                        Object {
                              "cursor": "pointer",
                              "fontSize": 16,
                              "left": 3,
                              "position": "relative",
                              "top": 4,
                            }
                  }
                  type="filter"
            />
            <div
                  onClick={[Function]}
                  style={
                        Object {
                              "display": "block",
                              "top": "auto10",
                            }
                  }
                  styleName="filter-elem"
            >
                  <CheckboxGroup
                        defaultValue={
                              Array [
                                    "received",
                                    "receiving",
                                    "running",
                                    "done",
                                    "waiting",
                                  ]
                        }
                        onChange={[Function]}
                        options={Array []}
                        prefixCls="ant-checkbox-group"
                        value={
                              Array [
                                    "received",
                                    "receiving",
                                    "running",
                                    "done",
                                    "waiting",
                                  ]
                        }
                  >
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="received"
                              >
                                    received
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="receiving"
                              >
                                    receiving
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="running"
                              >
                                    running
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="done"
                              >
                                    done
                              </Checkbox>
                        </p>
                        <p>
                              <Checkbox
                                    indeterminate={false}
                                    prefixCls="ant-checkbox"
                                    value="waiting"
                              >
                                    waiting
                              </Checkbox>
                        </p>
                  </CheckboxGroup>
                  <p
                        styleName="cut-off"
                  />
                  <a
                        onClick={[Function]}
                        style={
                              Object {
                                    "float": "left",
                                  }
                        }
                  >
                        OK
                  </a>
                  <a
                        onClick={[Function]}
                        style={
                              Object {
                                    "float": "right",
                                  }
                        }
                  >
                        Reset
                  </a>
            </div>
      </div>,
          "accessor": "messageStatus",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
          "width": 80,
        },
        Object {
          "Cell": [Function],
          "Header": "Arrival Time",
          "accessor": "arrivalTime",
          "minWidth": "auto",
          "resizable": false,
          "style": Object {
            "textAlign": "center",
          },
          "width": 120,
        },
      ]
    }
    data={Array []}
    defaultExpanded={Object {}}
    defaultFilterMethod={[Function]}
    defaultFiltered={Array []}
    defaultPageSize={11}
    defaultResized={Array []}
    defaultSortDesc={false}
    defaultSortMethod={[Function]}
    defaultSorted={Array []}
    expanded={Object {}}
    expanderDefaults={
      Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      }
    }
    filterable={false}
    freezeWhenExpanded={false}
    getLoadingProps={[Function]}
    getNoDataProps={[Function]}
    getPaginationProps={[Function]}
    getProps={[Function]}
    getResizerProps={[Function]}
    getTableProps={[Function]}
    getTbodyProps={[Function]}
    getTdProps={[Function]}
    getTfootProps={[Function]}
    getTfootTdProps={[Function]}
    getTfootTrProps={[Function]}
    getTheadFilterProps={[Function]}
    getTheadFilterThProps={[Function]}
    getTheadFilterTrProps={[Function]}
    getTheadGroupProps={[Function]}
    getTheadGroupThProps={[Function]}
    getTheadGroupTrProps={[Function]}
    getTheadProps={[Function]}
    getTheadThProps={[Function]}
    getTheadTrProps={[Function]}
    getTrGroupProps={[Function]}
    getTrProps={[Function]}
    groupedByPivotKey="_groupedByPivot"
    indexKey="_index"
    loading={false}
    loadingText="Loading..."
    manual={true}
    nestingLevelKey="_nestingLevel"
    nextText="Next"
    noDataText="No rows found"
    ofText="of"
    onExpandedChange={[Function]}
    onFetchData={[Function]}
    onPageChange={[Function]}
    onPageSizeChange={[Function]}
    onSortedChange={[Function]}
    originalKey="_original"
    pageSizeOptions={
      Array [
        5,
        10,
        15,
        20,
        30,
        40,
        50,
      ]
    }
    pageText="Page"
    pages={1}
    pivotDefaults={Object {}}
    pivotIDKey="_pivotID"
    pivotValKey="_pivotVal"
    previousText="Previous"
    resizable={true}
    rowsText="rows"
    showPageJump={true}
    showPageSizeOptions={true}
    showPagination={true}
    showPaginationBottom={true}
    showPaginationTop={false}
    sortable={false}
    style={
      Object {
        "height": "475px",
        "maxHeight": "475px",
      }
    }
    subRowsKey="_subRows"
  />
  <p
    style={
      Object {
        "lineHeight": "40px",
      }
    }
  >
    Total number 
    1
  </p>
</div>
`;
